web3.js/src/connection.ts

```diff
@@ -2865,39 +2909,72 @@ export class Connection {
       }
     });
 
-    let timeoutMs = this._confirmTransactionInitialTimeout || 60 * 1000;
-    switch (subscriptionCommitment) {
-      case 'processed':
-      case 'recent':
-      case 'single':
-      case 'confirmed':
-      case 'singleGossip': {
-        timeoutMs = this._confirmTransactionInitialTimeout || 30 * 1000;
-        break;
+    const checkBlockHeight = async () => {
+      const blockHeight = await this.getBlockHeight(commitment);
+      return blockHeight;
+    };
+
+    const expireTx = new Promise((resolve, reject) => {
+      if (typeof strategy === 'string') {
+        console.log('timeout');
+        let timeoutMs = this._confirmTransactionInitialTimeout || 60 * 1000;
+        switch (subscriptionCommitment) {
+          case 'processed':
+          case 'recent':
+          case 'single':
+          case 'confirmed':
+          case 'singleGossip': {
+            timeoutMs = this._confirmTransactionInitialTimeout || 30 * 1000;
+            break;
+          }
+          // exhaust enums to ensure full coverage
+          case 'finalized':
+          case 'max':
+          case 'root':
+        }
+        if (done) clearTimeout();
```

Comment:
I'll work on fixing this now. Thanks!

> This is one of the reasons that your test doesn't pass right now.

Which test is failing? When I run `yarn test` locally all the tests pass. I'd like to figure out what would be causing the discrepancy.

Reply:
I fully rebased and found a couple of things I need to fix for _you_ before this test will pass on the very latest.

Reply:
K. I made it so that you can mock an RPC message that never arrives.

```
await mockRpcMessage({
  method: 'signatureSubscribe',
  params: [blockHeightSignature, {commitment: 'finalized'}],
  result: new Promise(() => {}), // Make the RPC never respond,
});
```

Rebase after #24920 is merged (minutes from now) and you should be able to use that in all three of these new tests.

Reply:
Ah, okay

Reply:
Done! Rebase now.